File: 242.valid-anagram.py
#
# @lc app=leetcode id=242 lang=python3
#
# [242] Valid Anagram
#

# @lc code=start
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hash_record = [0] * 26
        for i in s:
            hash_record[ord(i) - ord('a')] += 1
        for i in t:
            hash_record[ord(i) - ord('a')] -= 1
        for i in hash_record:
            if i == 0:
                None
            else:return False
        return True
                
# @lc code=end

File: 242.valid-anagram.js
/*
 * @lc app=leetcode id=242 lang=javascript
 *
 * [242] Valid Anagram
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
   const hash_record = new Array(26).fill(0);
   for(let i=0;i<s.length;i++){
    hash_record['z'.charCodeAt('A')-s[i].charCodeAt('A')] += 1;
   }
   for (let i=0;i<t.length;i++){
    hash_record['z'.charCodeAt('A')-t[i].charCodeAt('A')] -= 1;
   }
   console.log(hash_record);
   for (let i=0;i<hash_record.length;i++){
    if (hash_record[i] !== 0){
        return false;
    }
   }
   return true;
};
// @lc code=end

